packages/amplify_authenticator/test/ui/main_test.dart

```diff
@@ -0,0 +1,229 @@
+  usernameWithAttributes(usernameWithAttributesConfig);
+
+  final String config;
+  const TestConfig(this.config);
+}
+
+enum ScreenGeometry {
+  // Geometry based off of an iPhone SE 2022
+  mini(size: Size(750, 1334), pixelRatio: 2),
+
+  // Geometry based off of an iPhone 14
+  mobile(size: Size(1170, 2532), pixelRatio: 3),
+
+  // Geometry based off of an iPad 10
+  tablet(size: Size(1620, 2160), pixelRatio: 2),
+
+  // Geometry based off of a Macbook Air/Pro 13" (2020)
+  laptop(size: Size(2560, 1600), pixelRatio: 2),
+
+  // Geometry based off of an ultra HD monitor
+  desktop(size: Size(2560, 1440), pixelRatio: 1);
+
+  final Size size;
+
+  final double pixelRatio;
+
+  const ScreenGeometry({required this.size, required this.pixelRatio});
+}
+
+enum TestTheme {
+  defaultMaterial,
+  highContrast,
+  customSwatch,
+  custom,
+}
+
+/// Adds [lightTheme] and [darkTheme] getters, which cannot be properties since
+/// that are not constants.
+extension TestThemeX on TestTheme {
+  ThemeData get lightTheme {
+    switch (this) {
+      case TestTheme.defaultMaterial:
+        return ThemeData.light();
+      case TestTheme.highContrast:
+        return ThemeData.from(
+          colorScheme: const ColorScheme.highContrastLight(),
+        );
+      case TestTheme.customSwatch:
+        return ThemeData.from(
+          colorScheme: ColorScheme.fromSwatch(
+            primarySwatch: Colors.red,
+            backgroundColor: Colors.white,
+          ),
+        ).copyWith(
+          indicatorColor: Colors.red,
+        );
+      case TestTheme.custom:
+        return ThemeData.light().copyWith(
+          tabBarTheme: const TabBarTheme(
+            labelColor: Colors.amber,
+          ),
+          indicatorColor: Colors.pink,
+        );
+    }
+  }
+
+  ThemeData get darkTheme {
+    switch (this) {
+      case TestTheme.defaultMaterial:
+        return ThemeData.dark();
+      case TestTheme.highContrast:
+        return ThemeData.from(
+          colorScheme: const ColorScheme.highContrastDark(),
+        );
+      case TestTheme.customSwatch:
+        return ThemeData.from(
+          colorScheme: ColorScheme.fromSwatch(
+            primarySwatch: Colors.red,
+            backgroundColor: Colors.black,
+            brightness: Brightness.dark,
+          ),
+        );
+      case TestTheme.custom:
+        return ThemeData.dark().copyWith(
+          tabBarTheme: const TabBarTheme(
+            labelColor: Colors.amber,
+          ),
+          indicatorColor: Colors.pink,
+        );
+    }
+  }
+}
+
+void main() {
+  final binding = TestWidgetsFlutterBinding.ensureInitialized();
+
+  setUp(() async {
+    // flutter's default test font does not accurately reflect font size,
+    // which leads to pixel over flows.
+    await loadAppFonts();
+
+    // default to mobile layout
+    binding.window.devicePixelRatioTestValue = ScreenGeometry.mobile.pixelRatio;
+    binding.window.physicalSizeTestValue = ScreenGeometry.mobile.size;
```

Comment:
Why are these needed if they get overridden below?

Reply:
Good catch. I had added it because I wasn't planning on adding ScreenGeometry as one of the dimensions for the theme tests, but changed my mind and added it. Updated.